src/history/hs.c

```diff
@@ -1018,7 +1026,7 @@ __wt_find_hs_upd(WT_SESSION_IMPL *session, WT_ITEM *key, uint64_t recno, WT_UPDA
               hs_cursor, &hs_stop_ts_tmp, &durable_timestamp_tmp, &upd_type_full, hs_value));
             upd_type = (uint8_t)upd_type_full;
         }
-
+        F_CLR(session, WT_SESSION_RESOLVING_MODIFY);
```

Comment:
Yes, the change is as per the discussion. The same clear statement is required in case of an error.